facebookads/test/integration.py

```diff
@@ -46,35 +46,70 @@ class FacebookAdsTestCase(unittest.TestCase):
     TEST_ID = str(int(time.time()) % 1000)
     TEST_IMAGE_PATH = os.path.join(os.path.dirname(__file__), 'test.png')
 
-    @classmethod
-    def new_test_ad_creative(cls):
-        creative = objects.AdCreative(
-            parent_id=cls.TEST_ACCOUNT.get_id_assured()
-        )
+    def setUp(self):
+        super(FacebookAdsTestCase, self).setUp()
+        self.remote_objects = []
+
+    def tearDown(self):
+        self.delete_remote_objects()
+        super(FacebookAdsTestCase, self).tearDown()
+
+    def delete_remote_objects(self):
+        """
+        Delete accumulated remote objects.
+        """
+        for o in reversed(self.remote_objects):
+            if o.Field.id in o and o.get_id() is not None:
+                try:
+                    o.remote_delete()
+                except Exception:
+                    if isinstance(o, objects.AdImage):
+                        # AdImages are often reused automatically since the
+                        # hash is unique. They can't be deleted until all the
+                        # references are cleaned up.
+                        pass
+                    else:
+                        # Error out the test if an object can't be deleted,
+                        # since this probably indicates a coding issue in the
+                        # tests.
+                        raise
+
+    def delete_me(self, obj):
+        """
+        Prepare to delete obj in tearDown.
+        """
+        self.remote_objects.append(obj)
+        return obj
+
+    def new_test_ad_creative(self):
+        creative = self.delete_me(objects.AdCreative(
+            parent_id=self.TEST_ACCOUNT.get_id_assured(),
+        ))
```

Comment:
Sorry, this will sound atrociously picky, but let's make `delete_me()` not return anything (since it's not immediately clear from reading the test what I'm getting back). Instead, it can just perform a single action of appending to `self.remote_objects`.

The code will then look like this:
```
creative = objects.AdCreative(
    parent_id=...,
)
self.delete_in_teardown(creative)
```